teps/0044-decouple-task-composition-from-scheduling.md

```diff
@@ -860,55 +995,71 @@ Pros:
 Cons:
 * The line between this and a Pipeline seems very thin
 * New CRD to contend with
+* Will mean that both the Pipelines controller and this controller will need to share (or duplicate) a lot of
+  Pipeline logic
 
-### Rely on the affinity assistant
-
-In this approach we'd rely on [the affinity assistant](https://github.com/tektoncd/community/pull/318/files) to
-co-locate pods that use the same workspace.
-
-Cons:
-* Doesn't help us with the overhead of multiple pods
-* [TEP-0046](https://github.com/tektoncd/community/pull/318/files) explores shortcomings of this approach
-
-### Custom scheduler
-
-In this approach we use a custom scheduler to schedule our pods.
-
-Cons:
-* [@jlpetterson has explored this option](https://docs.google.com/document/d/1lIqFP1c3apFwCPEqO0Bq9j-XCDH5XRmq8EhYV_BPe9Y/edit#heading=h.18c0pv2k7d1a)
-  and felt it added more complexity without much gain (see also https://github.com/tektoncd/pipeline/issues/3052)
-    * For example the issue [#3049](https://github.com/tektoncd/pipeline/issues/3049)
-* Doesn't help us with the overhead of multiple pods (each Task would still be a pod)
-
-### Support other ways to share data (e.g. buckets)
+###### Custom Pipeline
 
-In this approach we could add more workspace types that support other ways of sharing data between pods, for example
-uploading to and downloading from s3. (See [#290](https://github.com/tektoncd/community/pull/290/files).)
+This approach is very similar to [Grouping CRD](#grouping-crd) but instead of creating a new type in the Tekton API,
+the same functionality is implemented as a custom task.
 
-[This is something we support for "linking" PipelineResources.](https://github.com/tektoncd/pipeline/blob/master/docs/install.md#configuring-pipelineresource-storage)
+In this approach we solve this problem by making a Custom Task that can run a Pipeline using whatever scheduling
+mechanism is preferred; this assumes the custom task is the ONLY way we support a different scheduling than
+Task to pod going forward (even if we pick a different solution it could make sense to implement it as a custom task
+first).
 
 Pros:
-* Easier out of the box support for other ways of sharing data
+* Doesn't change anything about our API
 
 Cons:
-* Uploading and downloading at the beginning and end of every Task is not as efficient as being able to share the same
-  disk
-* We'd need to define an extension mechanism so folks can use whatever backing store they want
-* Doesn't help us with the overhead of multiple pods (each Task would still be a pod)
+* If this custom task is widely adopted, could fork our user community (or we'd have to decide to promote it to a
```

Comment:
Isn't this one of the goal of the CustomTask, to experiment, so that we make sure we get the correct abstraction and if we feel it needs to get into the API, we bring it in ?